### PR TITLE
feat: Add RequestLimitMiddleware for Service.MaxRequestSize config

### PR DESCRIPTION
**Pull request by jinlinGuan**
fixes edgexfoundry/edgex-go#3612

Signed-off-by: Ginny Guan <ginny@iotechsys.com>

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)will update later once this PR merged
  <link to docs PR>

## Testing Instructions

1. build services from this branch and set the MaxRequestSize config to 1
2. send POST request with a request body over 1KB
3. verify the responses, for example,
```
curl -ki -X POST http://localhost:59881/api/v2/deviceprofile/uploadfile -F "file=@./test-profile.yml"
HTTP/1.1 413 Request Entity Too Large
Content-Type: application/json
Date: Tue, 19 Apr 2022 01:45:10 GMT
Content-Length: 98
Connection: close

{"apiVersion":"v2","message":"request size exceed Service.MaxRequestSize(1 KB)","statusCode":413}
```

### CODE REVIEW

**Review by cloudxxx8 — APPROVED** [review]
LGTM

File: bootstrap/handlers/httpserver.go [hunk 1]
```diff
@@ -142,3 +150,30 @@ func (b *HttpServer) BootstrapHandler(
...
 	return true
 }
+
+// RequestLimitMiddleware is a middleware function that limits the request body size to Service.MaxRequestSize in kilobytes
+func RequestLimitMiddleware(sizeLimit int64, lc logger.LoggingClient) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			switch r.Method {
+			case http.MethodPost, http.MethodPut, http.MethodPatch:
+				if sizeLimit > 0 && r.ContentLength > sizeLimit*1000 {
```
Comment (lenny-intel) [inline bootstrap/handlers/httpserver.go:160]: one KB is 1024, not 1000
```suggestion
				if sizeLimit > 0 && r.ContentLength > sizeLimit*1024 {
```

File: bootstrap/handlers/httpserver_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func TestRequestLimitMiddleware(t *testing.T) {
+	lc := logger.NewMockClient()
+	payload := make([]byte, 2000)
```
Comment (lenny-intel) [inline bootstrap/handlers/httpserver_test.go:23]:
```suggestion
	payload := make([]byte, 2048)
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: bootstrap/handlers/httpserver_test.go (hunk 2 above)
Comment (jinlinGuan) [reply bootstrap/handlers/httpserver_test.go]: fixed

File: bootstrap/handlers/httpserver.go (hunk 1 above)
Comment (jinlinGuan) [reply bootstrap/handlers/httpserver.go]: fixed, also open issue https://github.com/edgexfoundry/edgex-go/issues/3980 to fix `MaxEventSize` value.

**Pull request merged**